ui/src/main/webapp/app/common/navigation-service.js

```diff
@@ -30,13 +30,33 @@ angular.module('app').factory('navigation', ['$location', 'events', 'common', fu
         navigation.goTo("query");
     }
 
+    /**
+     * Change the page, keeping the parameters in the url
+     */
     navigation.goTo = function(pageName) {
         if(pageName && common.startsWith(pageName, "/")) {
             pageName = pageName.substr(1);
         }
         currentPage = pageName;
         $location.path('/' + pageName);
-        events.broadcast('routeChange', [currentPage]);
+    }
+
+    /**
+     * Update the url with new parameters, without changing the page
+     */
+    navigation.updateURL = function(graphIds) {
```

Comment:
Although this will work for graphIds, it would not work for other operation options. It would be really neat for this to be generic and work with any options set from the settings page.